# Aircraft Safety Analysis for Potential Expansion into Aviation Business

## Overview

# Business Problem

# Data Understanding

In [1]:
import pandas as pd 
import numpy as np 
import matplotlib.pyplot as plt 
import seaborn as sns

In [7]:
df_original = pd.read_csv('../Data/AviationData.csv', encoding='latin1', low_memory=False)
df_original.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 88889 entries, 0 to 88888
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      88889 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Latitude                34382 non-null  object 
 7   Longitude               34373 non-null  object 
 8   Airport.Code            50132 non-null  object 
 9   Airport.Name            52704 non-null  object 
 10  Injury.Severity         87889 non-null  object 
 11  Aircraft.damage         85695 non-null  object 
 12  Aircraft.Category       32287 non-null  object 
 13  Registration.Number     87507 non-null  object 
 14  Make                    88826 non-null

### Original Dataset

In [4]:
df.head()

Date  Year  Month           Location        Country Injury_Severity  \
0  1990-01-01  1990      1        ELKHART, IN  United States       Non-Fatal   
1  1990-01-02  1990      1         MILLIS, MA  United States       Non-Fatal   
2  1990-01-03  1990      1   APPLE VALLEY, CA  United States       Non-Fatal   
3  1990-01-03  1990      1  SLIPPERY ROCK, PA  United States       Non-Fatal   
4  1990-01-03  1990      1   APPLE VALLEY, CA  United States       Non-Fatal   

   Damage_Type Aircraft_Category     Make    Model  ...  Weather  \
0  Substantial          Airplane   cessna     150F  ...      VMC   
1  Substantial          Airplane   cessna      150  ...      VMC   
2        Minor          Airplane   cessna     172M  ...      VMC   
3  Substantial          Airplane  aeronca     11CC  ...      VMC   
4  Substantial          Airplane   mooney  MARK 22  ...      VMC   

  Fatal_Injuries_Fraction  Serious_Injuries_Fraction  Minor_Injuries_Fraction  \
0                     0.0                        0.0                      0.0   
1                     0.0                        0.0                      1.0   
2                     0.0                        0.0                      0.0   
3                     0.0                        0.0                      0.0   
4                     0.0                        0.0                      0.0   

   Uninjured_Fraction  Risk_Index  Plane_Size  Make_and_Model  \
0                 1.0         0.0       Small     cessna 150F   
1                 0.0         0.1       Small      cessna 150   
2                 1.0         0.0       Small     cessna 172M   
3                 1.0         0.0       Small    aeronca 11CC   
4                 1.0         0.0       Small  mooney MARK 22   

   Mean_Risk_Index  Model_Total_Accidents  
0         0.140244                    164  
1         0.101643                    439  
2         0.122319                    539  
3         0.058333                      6  
4         0.000000                      1  

[5 rows x 27 columns]

## Cleaned dataset

The cleaned data is filtered down to US-only accident data from 1990 to 2022. Null values in columns relevant to our analysis have been removed. The original dataset of 88889 entries has been condensed down to 37461 entries. See Dataclean.ipynb, stringcleaning.ipynb, and EDA.ipynb for more details on how the data was cleaned.

In [9]:
df_cleaned = pd.read_csv('../CSVs/data_cleaned_final.csv')
df_cleaned.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 37461 entries, 0 to 37460
Data columns (total 27 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Date                       37461 non-null  object 
 1   Year                       37461 non-null  int64  
 2   Month                      37461 non-null  int64  
 3   Location                   37459 non-null  object 
 4   Country                    37461 non-null  object 
 5   Injury_Severity            37461 non-null  object 
 6   Damage_Type                37461 non-null  object 
 7   Aircraft_Category          37461 non-null  object 
 8   Make                       37461 non-null  object 
 9   Model                      37461 non-null  object 
 10  Engines                    37461 non-null  float64
 11  Engine_Type                36420 non-null  object 
 12  Fatal_Injuries             37461 non-null  float64
 13  Serious_Injuries           37461 non-null  flo